### PR TITLE
Feature/header references

### PR DESCRIPTION
**Pull request by ralphdoe**
New validations added to references of headers. 

### Description

Added a new validation when references of headers point to parameters instead of headers components. 

### Motivation and Context

There's no ticket about this yet, this is specified in SWOS-72. 

### How Has This Been Tested?

Yes, added unit tests for this case. 

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [X] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [X] are not breaking changes.

### Documentation
- [X] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [X] My changes can and should be tested by unit and/or integration tests.
- [X] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [X] All new and existing tests passed.

### CODE REVIEW

File: src/plugins/validate-semantic/validators/oas3/refs.js [hunk 1]
```diff
@@ -29,3 +29,31 @@ export const validateOAS3RefsForRequestBodiesReferenceRequestBodyPositions = ()
...
+      return nodes.reduce((acc, node) => {
+        const value = node.node
+        const ref = value.$ref
+
+        if(!ref) {
+          return acc
+        }
+        
+        // Ignore external references
+        if (ref.startsWith("#/")) {
```
Comment (hkosova) [inline src/plugins/validate-semantic/validators/oas3/refs.js:46]: This check is not needed because it's covered by the next `if`.

File: test/unit/plugins/validate-semantic/oas3/refs.js [hunk 2]
```diff
@@ -206,4 +206,109 @@ describe("validation plugin - semantic - oas3 refs", () => {
...
+
+                }
+              }
+            }
+          }
+        },
+        components: {
+          headers: {
+            MyHeader: {
+              $ref: "#/components/headers/MyHeader"
```
Comment (hkosova) [inline test/unit/plugins/validate-semantic/oas3/refs.js:234]: Is this `$ref` supposed to be
```
$ref: "#/components/parameters/MyHeader"
```
?

File: test/unit/plugins/validate-semantic/oas3/refs.js [hunk 3]
```diff
@@ -206,4 +206,109 @@ describe("validation plugin - semantic - oas3 refs", () => {
...
```
Comment (hkosova) [inline test/unit/plugins/validate-semantic/oas3/refs.js:305]: Is this `$ref` supposed to be
```
$ref: "https://www.google.com/#/components/parameter/MyHeader"
```
?

File: test/unit/plugins/validate-semantic/oas3/refs.js [hunk 4]
```diff
@@ -206,4 +206,109 @@ describe("validation plugin - semantic - oas3 refs", () => {
...
+            }
+          }
+        }
+      }
+
+      return expectNoErrorsOrWarnings(spec)
+    })
+    
+    it("should return no errors when a header correctly references a remote component header", () => {
```
Comment (hkosova) [inline test/unit/plugins/validate-semantic/oas3/refs.js:282]:
```suggestion
    it("should return no errors for external $refs in response headers", () => {
```

File: test/unit/plugins/validate-semantic/oas3/refs.js [hunk 5]
```diff
@@ -206,4 +206,109 @@ describe("validation plugin - semantic - oas3 refs", () => {
...
+        .then(system => {
+          const allErrors = system.errSelectors.allErrors().toJS()
+          const firstError = allErrors[1]
+          expect(allErrors.length).toEqual(2)
+          expect(firstError.message).toEqual(`OAS3 response header $refs should point to #/components/headers/... and not #/components/parameters/...`)
+          expect(firstError.path).toEqual(["paths", "/foo", "get","responses","200", "headers", "X-MyHeader", "$ref"])
+        })
+    })
+    
+    it("should return no errors when a header correctly references a local component header", () => {
```
Comment (hkosova) [inline test/unit/plugins/validate-semantic/oas3/refs.js:250]: Consider changing "header" -> "response header" to avoid confusion with header parameters.
```suggestion
    it("should return no errors when a response header correctly references a local header component", () => {
```

File: test/unit/plugins/validate-semantic/oas3/refs.js [hunk 6]
```diff
@@ -206,4 +206,109 @@ describe("validation plugin - semantic - oas3 refs", () => {
       return expectNoErrorsOrWarnings(spec)
     })
   })
-})
+
+  describe("$refs for headers must reference a header by position", () => {
+    it("should return an error when a header incorrectly references a local parameter schema", () => {
```
Comment (hkosova) [inline test/unit/plugins/validate-semantic/oas3/refs.js:211]: Naming suggestions for clarity:
```suggestion
  describe("response header $refs should not point to parameters", () => {
    it("should return an error when a response header incorrectly references a local parameter component", () => {
```

File: src/plugins/validate-semantic/validators/oas3/refs.js (hunk 1 above)
Comment (ralphdoe) [reply src/plugins/validate-semantic/validators/oas3/refs.js:46]: Done!

File: test/unit/plugins/validate-semantic/oas3/refs.js (hunk 6 above)
Comment (ralphdoe) [reply test/unit/plugins/validate-semantic/oas3/refs.js:211]: Done!

File: test/unit/plugins/validate-semantic/oas3/refs.js (hunk 5 above)
Comment (ralphdoe) [reply test/unit/plugins/validate-semantic/oas3/refs.js:250]: Done!

File: test/unit/plugins/validate-semantic/oas3/refs.js (hunk 4 above)
Comment (ralphdoe) [reply test/unit/plugins/validate-semantic/oas3/refs.js:282]: Done!

File: test/unit/plugins/validate-semantic/oas3/refs.js (hunk 3 above)
Comment (ralphdoe) [reply test/unit/plugins/validate-semantic/oas3/refs.js:305]: Yes, it makes sense. Done.

File: test/unit/plugins/validate-semantic/oas3/refs.js (hunk 2 above)
Comment (ralphdoe) [reply test/unit/plugins/validate-semantic/oas3/refs.js]: Done!

File: test/unit/plugins/validate-semantic/oas3/refs.js [hunk 7]
```diff
@@ -206,4 +206,109 @@ describe("validation plugin - semantic - oas3 refs", () => {
...
+              }
+            }
+          }
+        },
+        components: {
+          headers: {
+            MyHeader: {
+              $ref: "#/components/parameters/MyHeader"
+            }
+          }
```
Comment (hkosova) [inline test/unit/plugins/validate-semantic/oas3/refs.js:236]: Suggestion: Add a dummy parameter component to avoid extra `$ref` resolution errors in this test.
```suggestion
          },
          parameters: {
            MyHeader: {}
          }
```

File: test/unit/plugins/validate-semantic/oas3/refs.js (hunk 3 above)
Comment (hkosova) [reply test/unit/plugins/validate-semantic/oas3/refs.js:305]: @ralphdoe did you commit this change?

File: test/unit/plugins/validate-semantic/oas3/refs.js (hunk 7 above)
Comment (ralphdoe) [reply test/unit/plugins/validate-semantic/oas3/refs.js:236]: Done!

File: test/unit/plugins/validate-semantic/oas3/refs.js (hunk 3 above)
Comment (ralphdoe) [reply test/unit/plugins/validate-semantic/oas3/refs.js:305]: Done!
